Whiskey/Functions/Find-WhiskeyPowerShellModule.ps1

```diff
@@ -41,46 +41,119 @@ function Find-WhiskeyPowerShellModule
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-    # If you want to upgrade the PackageManagement and PowerShellGet versions, you must also update:
-    # * appveyor.yml
-    # * PublishPowerShellModule task
-    Import-Module -Name 'PackageManagement' -RequiredVersion '1.4.7' -Global
-    Import-Module -Name 'PowerShellGet' -RequiredVersion '2.2.5' -Global
-
-    Write-WhiskeyDebug -Message ('{0}  {1} ->' -f $Name,$Version)
-    if( $Version )
+    Write-WhiskeyDebug '\Find-WhiskeyPowerShellModule\' -Indent
+
+    function Import
     {
-        $atVersionString = ' at version {0}' -f $Version
+        [CmdletBinding()]
+        param(
+            [Parameter(Mandatory)]
+            [String] $Name,
+
+            [Parameter(Mandatory)]
+            [Version] $MinimumVersion,
 
-        if( -not [Management.Automation.WildcardPattern]::ContainsWildcardCharacters($version) -and [Version]::TryParse($Version,[ref]$null) )
+            [Parameter(Mandatory)]
+            [Version] $MaximumVersion
+        )
+
+        Write-WhiskeyDebug "Importing module $($Name) $($MinimumVersion) <= $($MaximumVersion)."
+        $module = Get-Module -Name $Name
+        if( $module )
         {
-            $tempVersion = [Version]$Version
-            if( $TempVersion -and ($TempVersion.Build -lt 0) )
+            if( $module.Version -ge $MinimumVersion -and $module.Version -le $MaximumVersion )
             {
-                $Version = [Version]('{0}.{1}.0' -f $TempVersion.Major, $TempVersion.Minor)
+                Write-WhiskeyDebug "Module $($Name) $($module.Version) already imported."
+                return
             }
+
+            Write-WhiskeyDebug "Removing module $($Name) $($module.Version)."
+            $module | Remove-Module -Force
         }
+        
+        $importFailed = $true
+        try
+        {
+            Import-Module -Name $Name `
+                          -MinimumVersion $MinimumVersion `
+                          -MaximumVersion $MaximumVersion `
+                          -Global

```

Comment:
Do you need to add an `-ErrorAction Stop` here to ensure ally errors are terminating?

Reply:
No. There didn't used to be a `try`/`finally` block here. Importing PackageManagement was throwing an exception which wasn't getting back up to the user. It was failing silently and then later on you'd get an error that PackageManagement module couldn't be found. We don't want to stop or handle if there's an error. We want to be able to at least debug unhandled exceptions when importing.